### PR TITLE
Remove null checks, instruct JsonWriter to ignore them.

### PR DESCRIPTION
**Pull request by JakeWharton**
This also updates JsonWriter to handle the Boolean case which has been [upstreamed into Gson itself][1].

```
$ ./gradlew clean build connectedCheck | tail -3
BUILD SUCCESSFUL

Total time: 31.71 secs
```

 [1]: https://github.com/google/gson/pull/836

### CODE REVIEW

**Comment by JakeWharton** [comment]
Build failure looks like a flake.

**Comment by kattrali** [comment]
The build failure was a flake. Thanks for the patch, @JakeWharton.

File: src/main/java/com/bugsnag/android/Error.java [hunk 1]
```diff
@@ -75,38 +74,16 @@ public void toStream(@NonNull JsonStream writer) throws IOException {
...
-            if(config.sendThreads) {
-                writer.name("threads").value(new ThreadState(config));
-            }
+            writer.name("app").value(appData);
+            writer.name("appState").value(appState);
+            writer.name("device").value(deviceData);
+            writer.name("deviceState").value(deviceState);
+            writer.name("breadcrumbs").value(breadcrumbs);
+            writer.name("groupingHash").value(groupingHash);
+            writer.name("threads").value(new ThreadState(config));
```
Comment (ndurell) [inline src/main/java/com/bugsnag/android/Error.java]: Sorry I meant to comment earlier...but do you want to always send this thread state object even if sendThreads is false?
Is this an option that is no longer used?

Comment (JakeWharton) [inline src/main/java/com/bugsnag/android/Error.java]: Yep. Looks like an accident. I'll send another PR in a few.

Comment (JakeWharton) [inline src/main/java/com/bugsnag/android/Error.java]: https://github.com/bugsnag/bugsnag-android/pull/106

**Pull request merged**